tools/topology/topology2/include/components/smart_amp.conf

```
@@ -0,0 +1,107 @@
+#
+# Common widget smart_amp
+#
+# A generic smart_amp widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.smart_amp.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.smart_amp."N" {
+# 		index		1
+# 		type		"effect"
+#	}
+#
+# Where N is the unique instance number for the smart_amp object within the same alsaconf node.
+
+Class.Widget."smart_amp" {
+	#
+	# Pipeline ID for the smart_amp widget object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# smart_amp object instance
+	#
+	DefineAttribute."instance" {}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#
+	# smart_amp widget UUID
+	#
+	DefineAttribute."uuid" {
+		type	"string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."is_pages" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	# Attribute categories
+	attributes {
+		#
+		# The smart_amp widget name would be constructed using the index and instance attributes.
+		# For ex: "smart_amp.1.1" or "smart_amp.10.2" etc.
+		#
+		!constructor [
+			"index"
+			"instance"
+		]
+
+		#
+		# mandatory attributes that must be provided when the class is instantiated
+		#
+		!mandatory [
+			"no_pm"
+			"uuid"
+			"process_type"
+		]
+
+		#
+		# immutable attributes cannot be modified in the object instance
+		#
+		!immutable [
+			"uuid"
+			"type"
+		]
+
+		#
+		# deprecated attributes should not be added in the object instance
+		#
+		!deprecated [
+			"preload_count"
+		]
+
+		#
+		# smart_amp widget objects instantiated within the same alsaconf node must have unique
+		# instance attribute
+		#
+		unique	"instance"
+	}
+
+	# Default attribute values for smart_amp widget
+
+	# uuid: 167a961e-8ae4-11ea-89f1-000c29ce1635
+	uuid			"1e:96:7a:16:e4:8a:ea:11:89:f1:00:0c:29:ce:16:35"
+	type			"effect"
+	no_pm 			"true"
```

what is that?

---

I don't know what is either but I know this would be much better to avoid double and triple negations:
```suggestion
	pm 			"false"
```
(I hope the default value can be either "true" or "false")

---

sorry this cannot be changed. no_pm is an existing terminology used by ALSA topology and must be set to true for all widgets